### PR TITLE
PMM-7296: List Alert Rules pagination

### PR DESCRIPTION
**Pull request by oter**
https://jira.percona.com/browse/PMM-7296

### CODE REVIEW

File: api/managementpb/ia/pagination.proto [hunk 1]
```diff
@@ -0,0 +1,21 @@
+syntax = "proto3";
+
+package ia.v1beta1;
+
+option go_package = "api/managementpb/ia;iav1beta1";
+
+// PaginationRequest represents request values for page by page pagination.
+message PaginationRequest {
```
Comment (artemgavrilov) [inline api/managementpb/ia/pagination.proto:8]: Typically `*Request/Response` are top-level entities, used as parameter/response for gRPC methods. Probably we should think about naming.

File: api/managementpb/ia/pagination.proto [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  int32 page = 2;
+}
+
+// PaginationResponse represents response values for page by page pagination.
+message PaginationResponse {
+  // Total number of results.
+  int32 total_items = 1;
+  // Total number of pages.
+  int32 total_pages = 2;
+}
```
Comment (artemgavrilov) [inline api/managementpb/ia/pagination.proto:21]: How client should understand what page index to pass to the next request? Calculate it each time form `page_size` and total returned items?

File: api/managementpb/ia/pagination.proto [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+package ia.v1beta1;
+
+option go_package = "api/managementpb/ia;iav1beta1";
+
+// PaginationRequest represents request values for page by page pagination.
+message PaginationRequest {
+  // Maximum number of results per page.
+  int32 page_size = 1;
+  // Index of the requested page, starts from 0.
+  int32 page = 2;
```
Comment (artemgavrilov) [inline api/managementpb/ia/pagination.proto:12]: Typically `cursor` gives a more slick UX than `offset` approach.  Maybe we can think about it?

Comment (Dasio) [reply api/managementpb/ia/pagination.proto:12]: Cursor is better also performance-wise but I don't think it is suitable for us. There won't be so many records >10000.
Also implementation is based on UI and there are pages. With cursor approach we can't go to exact page, just to the next.

**Comment by AlekSi** [comment]
(PR should not be a draft if reviews are already requested)

File: api/managementpb/ia/pagination.proto (hunk 1 above)
Comment (oter) [reply api/managementpb/ia/pagination.proto]: naming reconsidered

File: api/managementpb/ia/pagination.proto (hunk 2 above)
Comment (oter) [reply api/managementpb/ia/pagination.proto]: No need to calculate, the user chooses the page he wants to go to and this is a page index that passed in the request. 
The FE part knows the number of pages available from the response, so the appropriate number of page links is shown.

Comment (artemgavrilov) [reply api/managementpb/ia/pagination.proto]: Ok, thanks for the explanation!

File: api/managementpb/ia/pagination.proto (hunk 3 above)
Comment (artemgavrilov) [reply api/managementpb/ia/pagination.proto]: Yes, my message was about the representation approach, but it's out of code review scope.

**Review by artemgavrilov — APPROVED** [review]

**Review by Dasio — APPROVED** [review]

File: api/managementpb/ia/pagination.proto [hunk 4]
```diff
@@ -0,0 +1,21 @@
+syntax = "proto3";
+
+package ia.v1beta1;
+
+option go_package = "api/managementpb/ia;iav1beta1";
+
+// Page represents page request values for pagination.
+message Page {
```
Comment (BupycHuk) [inline api/managementpb/ia/pagination.proto:8]: maybe `PageParams` fits better here

**Review by BupycHuk — COMMENTED** [review]
just a few comments about namings

File: api/managementpb/ia/pagination.proto (hunk 4 above)
Comment (oter) [reply api/managementpb/ia/pagination.proto]: changed

File: api/managementpb/ia/pagination.proto [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  // Maximum number of results per page.
+  int32 size = 1;
```
[the remark this answers is not in the document]
Comment (oter) [reply api/managementpb/ia/pagination.proto]: renamed to `page_size`

**Review by BupycHuk — APPROVED** [review]

**Pull request merged**